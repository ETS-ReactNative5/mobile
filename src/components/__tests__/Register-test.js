import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { Register } from '../Register'

it('renders correctly', () => {
  const tree = renderer.create(
    <Register
      isFetching={false}
      register={() => {}}
      setField={() => {}}
      setError={() => {}}
      registration={{}} 
      setNavbarSettingsForPage={()=>{}}
      />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders errors', () => {
  const tree = renderer.create(
    <Register
      isFetching={false}
      register={() => {}}
      setField={() => {}}
      setError={() => {}}
      registration={{}}
      errorMessage={'Wrong!'} 
      setNavbarSettingsForPage={()=>{}}
      />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
