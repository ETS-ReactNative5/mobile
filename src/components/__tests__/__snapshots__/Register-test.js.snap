exports[`test renders correctly 1`] = `
<View
  style={undefined}>
  <ScrollView>
    <View
      style={undefined}>
      <View
        style={undefined}>
        <View
          style={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                undefined,
              ]
            }>
            User Name
          </Text>
        </View>
        <View
          style={undefined}>
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            keyboardType="default"
            onChangeText={[Function]}
            returnKeyType="done"
            secureTextEntry={false}
            style={undefined}
            underlineColorAndroid="white" />
        </View>
      </View>
      <View
        style={undefined}>
        <View
          style={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                undefined,
              ]
            }>
            Email address
          </Text>
        </View>
        <View
          style={undefined}>
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            keyboardType="email-address"
            onChangeText={[Function]}
            returnKeyType="done"
            secureTextEntry={false}
            style={undefined}
            underlineColorAndroid="white" />
        </View>
      </View>
      <View
        style={undefined}>
        <View
          style={undefined}>
          <View
            style={undefined}>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                ]
              }>
              Password
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }>
                Min 8 chars
            </Text>
          </View>
          <View
            style={
              Array [
                undefined,
                undefined,
              ]
            }>
            <TextInput
              autoCapitalize="none"
              autoCorrect={false}
              keyboardType="default"
              onChangeText={[Function]}
              returnKeyType="done"
              secureTextEntry={false}
              style={undefined}
              underlineColorAndroid="white" />
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }>
          <TouchableOpacity
            activeOpacity={0.2}
            disabled={undefined}
            onPress={[Function]}>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "padding": 20,
                }
              }>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "flex": 1,
                  }
                } />
              <TouchableOpacity
                activeOpacity={0.2}
                disabled={undefined}
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "borderColor": "#ddd",
                      "borderRadius": 5,
                      "borderWidth": 2,
                    },
                    Object {
                      "backgroundColor": "#1995ea",
                    },
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                  ]
                }>
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontFamily": "FontAwesome",
                        "fontSize": 12,
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      undefined,
                    ]
                  }>
                  
                </Text>
              </TouchableOpacity>
            </View>
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.8}
            onPress={[Function]}>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }>
              Show password
            </Text>
          </TouchableOpacity>
        </View>
      </View>
      <View
        style={undefined}>
        <View
          style={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                undefined,
              ]
            }>
            Real name
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                undefined,
                undefined,
              ]
            }>
              Optional
          </Text>
        </View>
        <View
          style={undefined}>
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            keyboardType="default"
            onChangeText={[Function]}
            returnKeyType="done"
            secureTextEntry={false}
            style={undefined}
            underlineColorAndroid="white" />
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
              undefined,
            ]
          }>
          We\'ll use this to give you credit in scientific papers, etc.
        </Text>
      </View>
      <View
        style={undefined}>
        <TouchableOpacity
          activeOpacity={0.2}
          disabled={undefined}
          onPress={[Function]}>
          <View
            style={
              Array [
                undefined,
                undefined,
              ]
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "flex": 1,
                }
              } />
            <TouchableOpacity
              activeOpacity={0.2}
              disabled={undefined}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "borderColor": "#ddd",
                    "borderRadius": 5,
                    "borderWidth": 2,
                  },
                  Object {
                    "backgroundColor": "white",
                  },
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                ]
              }>
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontFamily": "FontAwesome",
                      "fontSize": 12,
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }>
                
              </Text>
            </TouchableOpacity>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.8}
          onPress={[Function]}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                undefined,
                undefined,
              ]
            }>
            It’s okay to send me email occasionally.
          </Text>
        </TouchableOpacity>
      </View>
      <TouchableOpacity
        activeOpacity={0.5}
        disabled={true}
        onPress={[Function]}
        style={
          Array [
            undefined,
            undefined,
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={undefined}>
          Register
        </Text>
      </TouchableOpacity>
      <View
        style={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
              undefined,
            ]
          }>
          By signing up, I agree to Zooniverse\'s 
        </Text>
        <TouchableOpacity
          activeOpacity={0.5}
          onPress={[Function]}
          style={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                undefined,
                undefined,
              ]
            }>
            Privacy Policy
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  </ScrollView>
</View>
`;

exports[`test renders errors 1`] = `
<View
  style={undefined}>
  <ScrollView>
    <View
      style={undefined}>
      <View
        style={undefined}>
        <View
          style={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                undefined,
              ]
            }>
            User Name
          </Text>
        </View>
        <View
          style={undefined}>
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            keyboardType="default"
            onChangeText={[Function]}
            returnKeyType="done"
            secureTextEntry={false}
            style={undefined}
            underlineColorAndroid="white" />
        </View>
      </View>
      <View
        style={undefined}>
        <View
          style={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                undefined,
              ]
            }>
            Email address
          </Text>
        </View>
        <View
          style={undefined}>
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            keyboardType="email-address"
            onChangeText={[Function]}
            returnKeyType="done"
            secureTextEntry={false}
            style={undefined}
            underlineColorAndroid="white" />
        </View>
      </View>
      <View
        style={undefined}>
        <View
          style={undefined}>
          <View
            style={undefined}>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                ]
              }>
              Password
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }>
                Min 8 chars
            </Text>
          </View>
          <View
            style={
              Array [
                undefined,
                undefined,
              ]
            }>
            <TextInput
              autoCapitalize="none"
              autoCorrect={false}
              keyboardType="default"
              onChangeText={[Function]}
              returnKeyType="done"
              secureTextEntry={false}
              style={undefined}
              underlineColorAndroid="white" />
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }>
          <TouchableOpacity
            activeOpacity={0.2}
            disabled={undefined}
            onPress={[Function]}>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "padding": 20,
                }
              }>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "flex": 1,
                  }
                } />
              <TouchableOpacity
                activeOpacity={0.2}
                disabled={undefined}
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "borderColor": "#ddd",
                      "borderRadius": 5,
                      "borderWidth": 2,
                    },
                    Object {
                      "backgroundColor": "#1995ea",
                    },
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                  ]
                }>
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontFamily": "FontAwesome",
                        "fontSize": 12,
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      undefined,
                    ]
                  }>
                  
                </Text>
              </TouchableOpacity>
            </View>
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.8}
            onPress={[Function]}>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }>
              Show password
            </Text>
          </TouchableOpacity>
        </View>
      </View>
      <View
        style={undefined}>
        <View
          style={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                undefined,
              ]
            }>
            Real name
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                undefined,
                undefined,
              ]
            }>
              Optional
          </Text>
        </View>
        <View
          style={undefined}>
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            keyboardType="default"
            onChangeText={[Function]}
            returnKeyType="done"
            secureTextEntry={false}
            style={undefined}
            underlineColorAndroid="white" />
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
              undefined,
            ]
          }>
          We\'ll use this to give you credit in scientific papers, etc.
        </Text>
      </View>
      <View
        style={undefined}>
        <TouchableOpacity
          activeOpacity={0.2}
          disabled={undefined}
          onPress={[Function]}>
          <View
            style={
              Array [
                undefined,
                undefined,
              ]
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "flex": 1,
                }
              } />
            <TouchableOpacity
              activeOpacity={0.2}
              disabled={undefined}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "borderColor": "#ddd",
                    "borderRadius": 5,
                    "borderWidth": 2,
                  },
                  Object {
                    "backgroundColor": "white",
                  },
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                ]
              }>
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontFamily": "FontAwesome",
                      "fontSize": 12,
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }>
                
              </Text>
            </TouchableOpacity>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.8}
          onPress={[Function]}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                undefined,
                undefined,
              ]
            }>
            It’s okay to send me email occasionally.
          </Text>
        </TouchableOpacity>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            undefined,
            undefined,
          ]
        }>
        Wrong!
      </Text>
      <TouchableOpacity
        activeOpacity={0.5}
        disabled={true}
        onPress={[Function]}
        style={
          Array [
            undefined,
            undefined,
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={undefined}>
          Register
        </Text>
      </TouchableOpacity>
      <View
        style={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
              undefined,
            ]
          }>
          By signing up, I agree to Zooniverse\'s 
        </Text>
        <TouchableOpacity
          activeOpacity={0.5}
          onPress={[Function]}
          style={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                undefined,
                undefined,
              ]
            }>
            Privacy Policy
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  </ScrollView>
</View>
`;
